packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,863 @@
+              );
+            }
+            const provider = searchParams.get(
+              "provider_name"
+            ) as BuiltinOAuthProviderNames | null;
+            if (!provider || !builtinOAuthProviderNames.includes(provider)) {
+              throw new Error(`invalid provider_name: ${provider}`);
+            }
+            const redirectUrl = `${this._authRoute}/oauth/callback`;
+            const pkceSession = await this.core.then((core) =>
+              core.createPKCESession()
+            );
+            return redirect(
+              pkceSession.getOAuthUrl(
+                provider,
+                redirectUrl,
+                `${redirectUrl}?isSignUp=true`
+              ),
+              {
+                headers: new Headers({
+                  "Set-Cookie": cookie.serialize(
+                    this.options.pkceVerifierCookieName,
+                    pkceSession.verifier,
+                    { httpOnly: true }
+                  ),
+                }),
+              }
+            );
+          }
+
+          case "oauth/callback": {
+            if (!onOAuthCallback) {
+              throw new Error(
+                `'onOAuthCallback' auth route handler not configured`
+              );
+            }
+            const error = searchParams.get("error");
+            if (error) {
+              const desc = searchParams.get("error_description");
+              return onOAuthCallback({
+                error: new Error(error + (desc ? `: ${desc}` : "")),
+              });
+            }
+            const code = searchParams.get("code");
+            const isSignUp = searchParams.get("isSignUp") === "true";
+            const verifier =
+              parseCookies(req)[this.options.pkceVerifierCookieName];
+            if (!code) {
+              return onOAuthCallback({
+                error: new Error("no pkce code in response"),
+              });
+            }
+            if (!verifier) {
+              return cbCall(onOAuthCallback, {
+                error: new Error("no pkce verifier cookie found"),
+              });
+            }
+            let tokenData: TokenData;
+            try {
+              tokenData = await (await this.core).getToken(code, verifier);
+            } catch (err) {
+              return cbCall(onOAuthCallback, {
+                error: err instanceof Error ? err : new Error(String(err)),
+              });
+            }
+            const headers = new Headers();
+            headers.append(
+              "Set-Cookie",
+              cookie.serialize(
+                this.options.authCookieName,
+                tokenData.auth_token,
+                { httpOnly: true, sameSite: "lax", path: "/" }
+              )
+            );
+            headers.append(
+              "Set-Cookie",
+              cookie.serialize(this.options.pkceVerifierCookieName, "", {
+                maxAge: 0,
+              })
+            );
```

Comment:
I think all 'Set-Cookie' headers should have `path: "/"`.

Reply:
``If the server omits the Path attribute, the user agent will use the "directory" of the request-uri's path component as the default value.`` 

Hm I agree